src/components/PersonalInformationForm/FormValidations.js

```diff
@@ -59,8 +60,13 @@ const validations = {
 
   birthday: {
     custom: {
-      isValid: (value) => !value || !isNaN(new Date(value?.split("-").reverse().join("-"))),
-      message: "Selecione uma data de aniversário",
+      isValid: (value) => {
+        if(!value) return false;
+        if(dayjs(value).isAfter(dayjs(dayjs().subtract(18, "year")))) return false;

```

Comment:
Acho que essa logica acabou ficando um dificil de ler e também possui um bug. Acontece que o dayjs reconhece  `MM-DD-YYYY` como formato de input padrão, o que acabou fazendo com que, no objeto dayjs, os dias se invertessem com os meses. Sugeri uma mudança que resolve esse problema, e deixa a leitura da logica um pouco menos complicada:
```suggestion
        if(dayjs(value, "DD-MM-YYYY").isAfter(dayjs().subtract(18, "year"))) return false;
```
Isso deve resolver, sugiro testar os casos de borda essa minha sugestão e ver se está tudo funcionando ok :)